### DISCUSSION TITLE
Community-driven snippet collection

### DISCUSSION THREAD

**Discussion by antonreshetov** [massCodeIO/massCode#142] [Announcements]
I am excited to announce the launch of a snippet collection [site](https://masscode.io/snippets) where each of you can become a [contributor](https://github.com/massCodeIO/masscode.io/blob/master/README.md#contribution).  🚀

Let's make a community together 💪

![image](https://user-images.githubusercontent.com/17480987/185293724-e4c69777-7bec-4771-a6b7-a14036c5612d.png)

**Comment by antonreshetov** [comment]
Guys, we need a collective mind. 

I want the site to contain not only HTML & CSS, but also snippets for other languages. And I think the site needs some kind of system to organize snippets.

I looked at https://www.30secondsofcode.org/js/p/1 as for JS example and I don't like this systematization very much. They are Algorithm, Array, Browser, Date, Function, Math, Node, Object, which to me looks more like a language learning than a collection of snippets used every day. And the site looks more like a blog, it's a matter of taste of course, but I think it should be more like documentation. This of course does not change the fact that the guys did a great job.  

Each language will probably need to be adjusted to systematize, but there should be something in common. Do you have any idea where to start systematizing snippets?

**Reply by antonreshetov** [reply]
Temporarily still took this model of systematization, but so far nothing good comes to mind - https://masscode.io/snippets/javascript/array/

**Reply by vaimalaviya1233** [reply]
any guidelines to follow on submitting snippets for other languages?

**Reply by antonreshetov** [reply]
@vaimalaviya1233 There are no guidelines. We need to think about how to systemize this.

**Reply by vaimalaviya1233** [reply]
@antonreshetov could you please merge pr of python snippets. thanks.

**Comment by HANZ64** [comment]
Hi there,

If it's of any help, this is how my own JavaScript snippet organization looks like (so far).
![2022-08-26 13_00_04](https://user-images.githubusercontent.com/46830194/186899598-b5e0ccbc-1343-44fa-b28b-cd7366fc95bc.jpg)

**Comment by uhlhosting** [comment]
Any plans to add the whole Snippets Collection into a db.json to be importable ?

**Reply by CodingMax** [reply]
Can we still not do that in Early 2026?
I thought it was included, easily addable, or in some other way connected to the app?!

**Comment by Broderick-Westrope** [comment]
I'd love to see a **Golang Snippet Collection** using only the standard library. I think [GopherSnippets](https://gophersnippets.com/) looks like a good starting point. 

I'm still relatively new to the language, but I'm happy to get it started as I learn and take any feedback from better Gophers 🙂. If anyone has any other suggestions for sites with valuable snippets that could be added but don't have time to do it themselves I'd be happy to take a look.

**Reply by antonreshetov** [reply]
Any contribution to the [collection](https://masscode.io/snippets/) of snippets would be great.

**Comment by e-dreyer** [comment]
I would love a Python snippet library, especially for the iter and functools design patterns. Something like more_itertools which is also featured in the examples of the documentation for itertools.

**Comment by tzucker02** [comment]
I have a collection of snippets from Python, although most of them are not great, just everyday stuff I refer back to and use to build other scripts. Some are old ones from courses I took (or am taking on the MS level).
